integ/scripts/bash/run-e2e-automated.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+#
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# Used as the entry point to run the tests in an automated way. This expects the following env vars to be set:
+# USER_ACCEPTS_SSPL_FOR_RFDK_TESTS - Needs to be 'true' to use these tests
+# ROLE_ARN - The role that will be assumed to deploy the integ tests
+# ROLE_SESSION_NAME - A name for the assume role session
+# ROLE_EXTERNAL_ID - The external ID matching the assume role
+
+set -euo pipefail
+
+refreshcreds() {
+    echo "Refreshing credentials"
+    unset AWS_ACCESS_KEY_ID
+    unset AWS_SECRET_ACCESS_KEY
+    unset AWS_SESSION_TOKEN
+    creds=$(mktemp -d)/creds.json
+    AWS_STS_REGIONAL_ENDPOINTS=regional aws sts assume-role \
+        --role-arn $ROLE_ARN \
+        --role-session-name $ROLE_SESSION_NAME \
+        --external-id $ROLE_EXTERNAL_ID > $creds
+    export AWS_ACCESS_KEY_ID="$(cat ${creds} | grep "AccessKeyId" | cut -d'"' -f 4)"
+    export AWS_SECRET_ACCESS_KEY="$(cat ${creds} | grep "SecretAccessKey" | cut -d'"' -f 4)"
+    export AWS_SESSION_TOKEN="$(cat ${creds} | grep "SessionToken" | cut -d'"' -f 4)"
```

Comment:
Instead of writing to a file (`$(mktemp -d)/creds.json`), it is safer to just keep this in memory. Aside from other processes/agents being able to read the file, if the EBS volume is unencrypted and the underlying block is reused for another instance, a motivated attacker may be able to recover the credentials. The credentials are time-limited, but it is still good practice and doesn't hurt.

I believe you should be able to do this with:

```bash
export CREDS="$(AWS_STS_REGIONAL_ENDPOINTS=regional aws sts assume-role \
        --role-arn $ROLE_ARN \
        --role-session-name $ROLE_SESSION_NAME \
        --external-id $ROLE_EXTERNAL_ID)"
export AWS_ACCESS_KEY_ID="$(printenv CREDS | grep "AccessKeyId" | cut -d'"' -f 4)"
export AWS_SECRET_ACCESS_KEY="$(printenv CREDS | grep "SecretAccessKey" | cut -d'"' -f 4)"
export AWS_SESSION_TOKEN="$(printenv CREDS | grep "SessionToken" | cut -d'"' -f 4)"
# Clean up the env var
unset CREDS
```

We do this already when [configuring the database connection for the repo installer](https://github.com/aws/aws-rfdk/blob/f65591f1974f603de01fae20e3566b11c77cffa9/packages/aws-rfdk/lib/deadline/lib/database-connection.ts#L184-L187). The `printenv` is used to avoid having the credentials logged in shell history and observable in procfs.

Reply:
Updated